src/modules/book/book.controller.js

```diff
@@ -0,0 +1,98 @@
+const { validationResult } = require('express-validator');
+const bcrypt = require('bcrypt');
+
+const User = require('../user/user.model');
+const Auth = require('../auth/auth.model');
+const Book = require('./books.model');
+
+
+exports.createBook = async (req, res, next) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+        const error = new Error('validation failed, entered data is incorrect.');
+        error.statusCode = 422;
+        error.data = errors.array();
+        next(error);
+    }
+
+    const bookName = req.body.name;
+    const totalCopies = req.body.totalCopies;
+    const avaiableCopies = req.body.avaiableCopies;
+    try {
```

Comment:
Avoid using try-catch unless necessary; instead, raise a specific error with a statement like this: throw new UnAuthorizedError("You're not authorized to create a new book!");. Additionally, create a middleware to handle this error and return an appropriate response with status code